tests/test_knowledge_panels.py

```diff
@@ -19,114 +18,185 @@ def auto_activate_lang():
+        }
+    }
+
+
+def test_hunger_game_kp_label_with_value():
+    html = (
+        "<p><a href='https://hunger.openfoodfacts.org/questions?type=label&value_tag=en%3Aorganic'>"
+        "Answer robotoff questions about label en:organic</a></p>"
     )
-    assert hunger_game_kp(hunger_game_filter="brand", value="nestle") == {
+    assert KnowledgePanels(facet="label", value="en:organic").hunger_game_kp() == {
         "hunger-game": {
+            "title": "hunger-games",
             "elements": [
                 {
+                    "id": 0,
                     "element_type": "text",
-                    "text_element": {
-                        "html": html,
-                    },
+                    "text_element": {"html": html},
                 }
-            ]
+            ],
         }
     }
 
 
-def test_hunger_game_kp_label_with_value():
+def test_hunger_game_kp_with_all_tag_1():
     html = (
-        "<p><a href='https://hunger.openfoodfacts.org/questions?type=label&value_tag=organic'>"
-        "Answer robotoff questions about organic label</a></p>\n"
+        "<p><a href='https://hunger.openfoodfacts.org/questions?country=en%3Afrance&brand=lidl&type=category&value_tag=en%3Abeers'>"  # noqa: E501
+        "Answer robotoff questions about category en:beers</a></p>"
     )
-    assert hunger_game_kp(hunger_game_filter="label", value="organic") == {
+    assert KnowledgePanels(
+        facet="category",
+        value="en:beers",
+        sec_facet="brand",
+        sec_value="lidl",
+        country="france",
+    ).hunger_game_kp() == {
         "hunger-game": {
+            "title": "hunger-games",
             "elements": [
                 {
+                    "id": 0,
                     "element_type": "text",
-                    "text_element": {
-                        "html": html,
-                    },
+                    "text_element": {"html": html},
                 }
-            ]
+            ],
+        }
+    }
+
+
+def test_hunger_game_kp_with_all_tag_2():
+
+    html0 = (
+        "<p><a href='https://hunger.openfoodfacts.org/questions?country=en%3Abelgium&type=category&value_tag=en%3Acoffees'>"  # noqa: E501
+        "Answer robotoff questions about category en:coffees</a></p>"
+    )
+    html1 = (
+        "<p><a href='https://hunger.openfoodfacts.org/questions?country=en%3Abelgium&type=brand'>"
+        "Answer robotoff questions about brand</a></p>"
+    )
+    assert KnowledgePanels(
+        facet="brand",
+        sec_facet="category",
+        sec_value="en:coffees",
+        country="belgium",
+    ).hunger_game_kp() == {
+        "hunger-game": {
+            "title": "hunger-games",
+            "elements": [
+                {"id": 0, "element_type": "text", "text_element": {"html": html0}},
+                {"id": 1, "element_type": "text", "text_element": {"html": html1}},
+            ],
+        }
+    }
```

Comment:
@alexgarel `test_hunger_game_kp_with_all_tag_2 and 3` is failing due to change in  value `html0` `html1` while adding urls, in `knowledge_panels.py`

Reply:
Any suggestion?

Reply:
It's like sometimes `html0` became `html1` and vice-versa.

Reply:
here,
![image](https://user-images.githubusercontent.com/96618001/190189352-91edfe65-ab90-49d5-a31f-0e3244bcb14e.png)

Reply:
hum, indeed this is because we puts facets in a dict (which does not have a predictable order).

I would modify the function, instead of `for k, v in facets.items()` to `for k, v in sorted(facets.items())` for it's predictable.